stl/inc/xmemory

```diff
@@ -1231,10 +1230,9 @@ _CONSTEXPR20 void _Container_base12::_Orphan_all_unlocked_v3() noexcept {
     }
 
     // proxy allocated, drain it
-    for (auto& _Pnext = _Myproxy->_Myfirstiter; _Pnext; _Pnext = _Pnext->_Mynextiter) { // TRANSITION, VSO-1269037
+    for (auto& _Pnext = _Myproxy->_Myfirstiter; _Pnext; _Pnext = _Pnext->_Mynextiter) {
         _Pnext->_Myproxy = nullptr;
     }

```

Comment:
Should we instead:
```suggestion
    for (auto _Pnext = exchange(_Myproxy->_Myfirstiter, nullptr); _Pnext; _Pnext = _Pnext->_Mynextiter) {
        _Pnext->_Myproxy = nullptr;
    }
```
so we write `_Myproxy->_Myfirstiter` only once? (And to remove the subtlety here that confused us before?)

Reply:
Good idea, that makes the behavior a lot clearer. Done (with `_STD`).